src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ApproverCaseloadService.kt

```diff
@@ -3,24 +3,141 @@ package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.caseload
+    val nomisRecords = prisonerSearchApiClient.searchPrisonersByNomisIds(deliusNomisIds)
+
+    val prisonerRecord: List<Triple<PrisonerSearchPrisoner, OffenderDetail?, LicenceSummaryApproverView?>> =
+      nomisRecords.map {
+        Triple(it, it.getDeliusRecord(deliusRecords), it.getLicenceSummary(licences))
+      }
+
+    val caseload = prisonerRecord.map { (nomisRecord, deliusRecord, licenceSummary) ->
+      val releaseDate = when (licenceSummary) {
+        null -> nomisRecord.selectReleaseDate()
+        else -> licenceSummary.actualReleaseDate ?: licenceSummary.conditionalReleaseDate
+      }
+      ApprovalDetails(
+        deliusRecord = deliusRecord,
+        comUsernameOnLicence = licenceSummary?.comUsername,
+        licenceSummary = ApprovalLicenceSummary(
+          licenceId = licenceSummary?.licenceId,
+          name = "${nomisRecord.firstName} ${nomisRecord.lastName}".convertToTitleCase(),
+          prisonerNumber = nomisRecord.prisonerNumber,
+          submittedByFullName = licenceSummary?.submittedByFullName,
+          releaseDate = releaseDate,
+          urgentApproval = licenceSummary?.isDueToBeReleasedInTheNextTwoWorkingDays,
+          approvedBy = licenceSummary?.approvedByName,
+          approvedOn = licenceSummary?.approvedDate,
+          isDueForEarlyRelease = licenceSummary?.isDueForEarlyRelease,
+        ),
+      )
+    }
+
+    val comUsernames = caseload.mapNotNull { it.comUsernameOnLicence }
+    val coms = communityApiClient.getStaffDetailsByUsername(comUsernames)
+
+    val approvalCases = caseload.map {
+      ApprovalCase(
+        licenceId = it.licenceSummary?.licenceId,
+        name = it.licenceSummary?.name,
+        prisonerNumber = it.licenceSummary?.prisonerNumber,
+        submittedByFullName = it.licenceSummary?.submittedByFullName,
+        releaseDate = it.licenceSummary?.releaseDate,
+        urgentApproval = it.licenceSummary?.urgentApproval,
+        approvedBy = it.licenceSummary?.approvedBy,
+        approvedOn = it.licenceSummary?.approvedOn,
+        isDueForEarlyRelease = it.licenceSummary?.isDueForEarlyRelease,
+        probationPractitioner = findProbationPractitioner(it.deliusRecord, it.comUsernameOnLicence, coms),
+      )
+    }.sortedByDescending { it.releaseDate }
+
+    return approvalCases
   }
+}
 
-  companion object {
-    val log: Logger = LoggerFactory.getLogger(this::class.java)
+fun PrisonerSearchPrisoner.getDeliusRecord(deliusRecords: List<OffenderDetail>): OffenderDetail? {
+  return deliusRecords.find { it.otherIds.nomsNumber == this.prisonerNumber }
+}
+
+fun PrisonerSearchPrisoner.getLicenceSummary(licences: List<LicenceSummaryApproverView>): LicenceSummaryApproverView? {
+  val licenceSummaries = licences.filter { it.nomisId == this.prisonerNumber }
+  return if (licenceSummaries.size == 1) licenceSummaries.first() else licenceSummaries.find { it.licenceStatus != LicenceStatus.ACTIVE }
+}
+
+fun PrisonerSearchPrisoner.selectReleaseDate(): LocalDate? {
```

Comment:
**suggestion(non-blocking):** we have this logic on prisoner search prisoner, in lots of places now.. 

Would be good to extract this out and share maybe into its own module in `uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/prison`

Can see it in:
* LicenceFactory x 2
* EligibilityService

(There's an example in prisonerSearchService which looks similar where we fallback to releaseDate which probably should be CRD too)